src/panic.md

```diff
@@ -0,0 +1,51 @@
+# Panic
+
+Rust provides the ability to "panic" upon encountering a runtime error that is
+not part of a function's signature; such an error is typically not expected to
+be recoverable within the context in which the error is encountered.
+
+> The Standard Library provides this capability via the [`panic!` macro][macro-panic].
+
+Although it is not part of the core language, panics interact with several core
```

Comment:
I think I understand what this paragraph is getting at, but I keep feeling uncomfortable with it. It introduces a term "core language" which is never defined. What is the non-core language? Why are panics being documented here if they are not part of the language? What is the intent with stating this? 

I would say that panics *are* a part of Rust The Language, especially considering they are mentioned in various places like ABI strings, and the required interactions with those. I would lean towards not trying to claim a Rust implementation could just not support panics.

Reply:
I was trying to keep the distinctions clear that you mentioned [here](https://rust-lang.zulipchat.com/#narrow/stream/237824-t-lang.2Fdoc/topic/C-unwind.20and.20panic.20runtime.20documentation/near/284304845). (This was almost a year ago, so I understand if you don't remember writing that!) The intent of the paragraph is to indicate that _even though_ `panic` is a macro implemented in the Standard Library rather than something implemented entirely in the compiler, the compiler nevertheless must be aware of panicking as a capability of the runtime.

I'm not sure what would be a better way to indicate this. Perhaps the sentence above about the standard library could be modified to indicate that the macro itself is "magic" in some way, i.e., relies on a compiler intrinsic?

Reply:
How about directly saying there are multiple parts involved here? This could serve as a high-level introduction, and a brief introduction of what follows. Such as:

> The mechanism to start a panic is provided by the [`panic!` macro](../std/macro.panic.html) in the standard library. Panics interact with language constructs such as unwinding behavior defined by a [function ABI](items/functions.md#unwinding) or out-of-bounds [array indexing](expressions/array-expr.md#array-and-slice-indexing-expressions). A [*panic runtime*](#panic-runtimes) defined by the compiler controls how a panic is handled during runtime.

Reply:
@ehuss I've taken that as inspiration and tried something a little different, first listing how array-index and FFI ABI strings interact with panics, then mentioning `panic!` as the standard library's way to explicitly panic (i.e. rather than automatically panicking). How's the current text?

Reply:
Looks great, thanks!